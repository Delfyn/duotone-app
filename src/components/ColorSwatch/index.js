/* @flow */

import React from 'react';

type PropType = {
  className?: string,
  primary?: string,
  secondary?: string,
  onClick: Function,
  size?: number,
  rotation?: number,
};

const ColorSwatch = ({
  onClick,
  className = '',
  primary = '#9B9B9B',
  secondary = '#4A4A4A',
  size = 40,
  rotation = 45,
}: PropType) => (
  <svg
    className={className}
    onClick={() => onClick(primary, secondary)}
    width={size}
    height={size}
    viewBox={`0 0 ${size} ${size}`}
    xmlns="http://www.w3.org/2000/svg">
    <defs>
      <circle id="a" cx={size / 2} cy={size / 2} r={size / 2} />
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
      transform={`rotate(${rotation} ${size / 2} ${size / 2})`}>
      <mask id="b" fill="#fff">
        <use xlinkHref="#a" />
      </mask>
      <use fill={primary} xlinkHref="#a" />
      <path
        fill={secondary}
        mask="url(#b)"
        d={`M${size / 2} 0h${size / 2}v${size}H${size / 2}z`}
      />
    </g>
  </svg>
);

export default ColorSwatch;
