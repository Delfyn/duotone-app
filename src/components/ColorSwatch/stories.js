/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import ColorSwatch from './index';
import duotones from '../../lib/color-claim';

storiesOf('ColorSwatch', module)
  .add('without props', () => (
    <ColorSwatch onClick={action('clicked')} />
  ))
  .add('with different colors', () => (
    <div>
      <ColorSwatch
        primary="#D42D48"
        secondary="#243261"
        onClick={action('clicked')}
      />
      <ColorSwatch
        primary="#D3F178"
        secondary="#263661"
        onClick={action('clicked')}
      />
      <ColorSwatch
        primary="#FEFFFF"
        secondary="#D5E5E8"
        onClick={action('clicked')}
      />
    </div>
  ))
  .add('with color-claim', () => (
    <div>
      {duotones.map((colors, index) => (
        <ColorSwatch
          {...colors}
          className="mr1 mb1"
          onClick={action('clicked')}
          key={index}
        />
      ))}
    </div>
  ))
  .add('with 0 degree rotation', () => (
    <ColorSwatch
      primary="#D42D48"
      secondary="#243261"
      rotation={0}
      onClick={action('clicked')}
    />
  ))
  .add('with bigger size', () => (
    <ColorSwatch
      primary="#D42D48"
      secondary="#243261"
      size={200}
      onClick={action('clicked')}
    />
  ));
