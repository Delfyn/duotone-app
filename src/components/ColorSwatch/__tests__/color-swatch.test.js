import React from 'react';
import renderer from 'react-test-renderer';
import ColorSwatch from '../index';

test('ColorSwatch renders correctly', () => {
  const onClick = jest.fn();
  const component = renderer.create(
    <ColorSwatch primary="#ff0000" secondary="#00ff00" onClick={onClick} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(onClick).toBeCalled();
});

test('ColorSwatch renders different sizes correctly', () => {
  const onClick = jest.fn();
  const component = renderer.create(
    <ColorSwatch
      primary="#ff0000"
      secondary="#00ff00"
      onClick={onClick}
      size={64}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ColorSwatch renders rotations sizes correctly', () => {
  const onClick = jest.fn();
  const component = renderer.create(
    <ColorSwatch
      primary="#ff0000"
      secondary="#00ff00"
      onClick={onClick}
      size={64}
      rotate={-45}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
