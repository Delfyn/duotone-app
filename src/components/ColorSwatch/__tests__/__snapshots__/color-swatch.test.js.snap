// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwatch renders correctly 1`] = `
<svg
  className=""
  height={40}
  onClick={[Function]}
  viewBox="0 0 40 40"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <circle
      cx={20}
      cy={20}
      id="a"
      r={20}
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    transform="rotate(45 20 20)"
  >
    <mask
      fill="#fff"
      id="b"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
    <use
      fill="#ff0000"
      xlinkHref="#a"
    />
    <path
      d="M20 0h20v40H20z"
      fill="#00ff00"
      mask="url(#b)"
    />
  </g>
</svg>
`;

exports[`ColorSwatch renders correctly 2`] = `
<svg
  className=""
  height={40}
  onClick={[Function]}
  viewBox="0 0 40 40"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <circle
      cx={20}
      cy={20}
      id="a"
      r={20}
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    transform="rotate(45 20 20)"
  >
    <mask
      fill="#fff"
      id="b"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
    <use
      fill="#ff0000"
      xlinkHref="#a"
    />
    <path
      d="M20 0h20v40H20z"
      fill="#00ff00"
      mask="url(#b)"
    />
  </g>
</svg>
`;

exports[`ColorSwatch renders different sizes correctly 1`] = `
<svg
  className=""
  height={64}
  onClick={[Function]}
  viewBox="0 0 64 64"
  width={64}
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <circle
      cx={32}
      cy={32}
      id="a"
      r={32}
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    transform="rotate(45 32 32)"
  >
    <mask
      fill="#fff"
      id="b"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
    <use
      fill="#ff0000"
      xlinkHref="#a"
    />
    <path
      d="M32 0h32v64H32z"
      fill="#00ff00"
      mask="url(#b)"
    />
  </g>
</svg>
`;

exports[`ColorSwatch renders rotations sizes correctly 1`] = `
<svg
  className=""
  height={64}
  onClick={[Function]}
  viewBox="0 0 64 64"
  width={64}
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <circle
      cx={32}
      cy={32}
      id="a"
      r={32}
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
    transform="rotate(45 32 32)"
  >
    <mask
      fill="#fff"
      id="b"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
    <use
      fill="#ff0000"
      xlinkHref="#a"
    />
    <path
      d="M32 0h32v64H32z"
      fill="#00ff00"
      mask="url(#b)"
    />
  </g>
</svg>
`;
